cli/tests/test_executor.py

```
@@ -609,6 +609,31 @@ def test_extract_error_fallback(self, mock_which):
+    @patch("shutil.which")
+    def test_extract_error_http_403(self, mock_which):
+        mock_which.return_value = "/usr/bin/act"
+        executor = JobExecutor(logs_dir=Path("/tmp/localci-test"))
+        output = "fetching action\nreceived HTTP status: 403\nend"
+        extracted = executor._extract_error(output)
+        assert "403" in extracted
+
+    @patch("shutil.which")
+    def test_extract_error_ignores_401_false_positive(self, mock_which):
+        mock_which.return_value = "/usr/bin/act"
+        executor = JobExecutor(logs_dir=Path("/tmp/localci-test"))
```

Ruff (0.15.15)

[error] Probable insecure usage of temporary file or directory: "/tmp/localci-test"

(S108)